java-client/src/test/java/com/cumulocity/sdk/client/binary/BinaryApiImplTest.java

```diff
@@ -62,13 +62,28 @@ public void shouldUploadFileAsByteArray() throws SDKException {
         assertThat(mor, sameInstance(created));
     }
 
+    @Test
+    public void shouldUploadFileAsByteArrayWithCustomMediaType() throws SDKException {
+        // Given
+        byte[] binaryData = {1,2,4,6,6};
+        ManagedObjectRepresentation managedObjectRepresentation = new ManagedObjectRepresentation();
+        ManagedObjectRepresentation created = new ManagedObjectRepresentation();
+        when(restConnector.postFile(INVENTORY_BINARY_URL, managedObjectRepresentation, binaryData, MediaType.TEXT_PLAIN_TYPE, ManagedObjectRepresentation.class)).thenReturn(created);
+
+        // When
+        ManagedObjectRepresentation mor = binariesApi.uploadFile(managedObjectRepresentation, binaryData, MediaType.TEXT_PLAIN_TYPE);
+
+        // Then
+        assertThat(mor, sameInstance(created));
```

Comment:
minor: I know this test is similar to previous, but maybe we can improve it and just verify that `restConnector` called `postFile()` with specific parameters, because this is in fact indirectly tested in lines #71 and #77.

Reply:
As you correctly noticed this assertion (calling postFile with correct parameters) is verified by creating specific mock at test setup, so I see no point in verifying it again.
If the method would be called with different parameters or not called at all, test will fail anyway.

Reply:
I was thinking about removing previous assertion and setup from mentioned lines 70, 71, 77 and only use directly `verify()`